### PR TITLE
Plotting/DataFrame/Webbrowser: Moved DataFrameColumnProperty 

### PR DESCRIPTION
**Pull request by ResearchDaniel**
Moved DataFrameColumnProperty to the DataFrame module and added a json-converter for it in the web browser module

### CODE REVIEW

File: modules/dataframe/CMakeLists.txt [hunk 1]
```diff
@@ -34,13 +36,15 @@ set(SOURCE_FILES
...
 	src/io/jsonreader.cpp
+	src/io/json/dataframepropertyjsonconverter.cpp
 	src/processors/csvsource.cpp
 	src/processors/dataframeexporter.cpp
 	src/processors/dataframesource.cpp
 	src/processors/imagetodataframe.cpp
 	src/processors/syntheticdataframe.cpp
 	src/processors/volumesequencetodataframe.cpp
 	src/processors/volumetodataframe.cpp
+    src/properties/dataframeproperty.cpp
```
Comment (martinfalk) [inline modules/dataframe/CMakeLists.txt]: indentation

File: modules/dataframe/include/inviwo/dataframe/io/json/dataframepropertyjsonconverter.h [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+ * }
+ * @see DataFrameColumnProperty
+ *
+ * Usage example:
+ * \code{.cpp}
+ * DataFrameColumnProperty p;
+ * json j = p;
+ * \endcode
+ */
+IVW_MODULE_DATAFRAME_API void to_json(json& j, const DataFrameColumnProperty& p);
```
Comment (martinfalk) [inline modules/dataframe/include/inviwo/dataframe/io/json/dataframepropertyjsonconverter.h]: why the underscore?

File: modules/dataframe/include/inviwo/dataframe/io/json/dataframepropertyjsonconverter.h [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+ *
+ *********************************************************************************/
+
+#pragma once
+
+#include <inviwo/dataframe/properties/dataframeproperty.h>
+#include <modules/json/io/json/optionpropertyjsonconverter.h>
+#include <nlohmann/json.hpp>
+
+using json = nlohmann::json;
```
Comment (martinfalk) [inline modules/dataframe/include/inviwo/dataframe/io/json/dataframepropertyjsonconverter.h]: is this really necessary in the header file?

File: modules/dataframe/src/io/json/dataframepropertyjsonconverter.cpp [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ *********************************************************************************/
+
+#include <inviwo/dataframe/io/json/dataframepropertyjsonconverter.h>
+
```
Comment (martinfalk) [inline modules/dataframe/src/io/json/dataframepropertyjsonconverter.cpp:32]: no additional includes necessary for the calls to `to_json()` and `from_json()` ?

File: modules/dataframe/include/inviwo/dataframe/io/json/dataframepropertyjsonconverter.h (hunk 2 above)
Comment (ResearchDaniel) [reply modules/dataframe/include/inviwo/dataframe/io/json/dataframepropertyjsonconverter.h]: Part of the json library

File: modules/dataframe/src/io/json/dataframepropertyjsonconverter.cpp (hunk 4 above)
Comment (ResearchDaniel) [reply modules/dataframe/src/io/json/dataframepropertyjsonconverter.cpp:32]: nope

**Comment by ResearchDaniel** [comment]
@martinfalk Fixed the minor comments

**Review by martinfalk — APPROVED** [review]

**Pull request merged**